### ISSUE TITLE
Linter does not catch MalformedPolicyDocument syntax error

### ISSUE THREAD

**Issue by DavidHospital** [aws-cloudformation/cfn-lint#2021]
*cfn-lint version: 0.25.0*

*Description of issue.*
`cfn-lint` fails to catch the following syntax error pre-deployment, which is caught by CloudFormation during deployment:
```
Syntax errors in policy. (Service: AmazonIdentityManagement; Status Code: 400; Error Code: MalformedPolicyDocument; Request ID: ccd05e19-aa60-4b26-bfcc-8fbafca40c61; Proxy: null)
```

The policy in question has the following template:
```yaml
APILambdaExecutionPolicy:
    Type: AWS::IAM::ManagedPolicy
    Properties:
      Description: Allows API Gateway triggered Lambdas access to service resources
      Path: /
      ManagedPolicyName:
        Fn::Sub: ${Project}-APILambdaExecutionPolicy-${AWS::Region}
      PolicyDocument:
        Version: '2012-10-17'
        Statement:
        - Effect: Allow
          Action:
          - dynamodb:*Get*
          - dynamodb:*Item*
          - dynamodb:Describe*
          - dynamodb:Query
          Resource:
          - Fn::GetAtt:
            - DynamoDB
            - Outputs.TaskDBTableARN
          - Fn::Sub: ${DynamoDB.Outputs.TaskDBTableARN}/*
          - Effect: Allow
            Action:
            - cloudwatch:PutMetricAlarm
            - cloudwatch:PutMetricData
            Resource:
            - '*'
          - Effect: Allow
            Action:
            - xray:PutTraceSegments
            - xray:PutTelemetryRecords
            Resource:
            - '*'
          - Effect: Allow
            Action:
            - sqs:GetQueueUrl
            - sqs:SendMessage
            Resource:
            - Fn::GetAtt:
              - SQS
              - Outputs.QueueARN
          - Effect: Allow
            Action:
            - sns:Publish
            Resource:
            - arn:aws:sns:*:*:*  
```

**Comment by PatMyron** [comment]
To clarify, is the syntax error for this template just the indentation?

**Comment by kddejong** [comment]
Included logic in rule E2507 that will now cover this.  That rule is still listed as experimental.

Since IAM policies are of type JSON we have to build additional logic to test these things.
